### PR TITLE
Intake-ifying osm

### PR DESCRIPTION
**Pull request by jsignell**
This PR:
 - moves osm-1billion
 - makes that the default notebook for the project
 - intake-ifies that notebook to allow download only when called
 - adds the infrastructure for intake test catalogs

Blocked on https://github.com/intake/intake-parquet/pull/11 and intake-parquet release.

### CODE REVIEW

File: osm/anaconda-project.yml [hunk 1]
```diff
@@ -50,10 +50,11 @@ env_specs:
     packages:
       - holoviews
       - intake
-      - intake-parquet
       - ipywidgets
       - jupyter
       - tqdm
+      - pip:
+        - git+https://github.com/intake/intake-parquet.git@jsignell/spatial#egg=intake_parquet
```
Comment (jbednar) [inline osm/anaconda-project.yml]: intake-parquet isn't on conda?

File: osm/osm-1billion.ipynb [hunk 2]
```diff
@@ -39,8 +39,25 @@
...
+    "cat = intake.open_catalog('./catalog.yml')\n",
+    "data_entry = cat.osm_one_billion\n",
+    "data_entry"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note that the file isn't downloaded yet. The following step will take some time:"
```
Comment (jbednar) [inline osm/osm-1billion.ipynb]: How do you know at this point whether it's been downloaded yet? Won't it be cached?  Seems like it should say instead that "Note that the first time this cell is executed, the file will take some time to download, but subsequent runs will skip that step".

File: test_data/osm/catalog.yml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+sources:
+  osm_one_billion:
+    description: Test data points to same fake osm-3billion file
```
Comment (jbednar) [inline test_data/osm/catalog.yml]: Same as what?  Try to reword in a way that makes sense when reading just this one file.

File: osm/anaconda-project.yml (hunk 1 above)
Comment (jsignell) [reply osm/anaconda-project.yml]: This points to a branch. The spatial isn't merged or released yet.

**Comment by jsignell** [comment]
There is a lingering issue here where the urlpath that is returned in https://github.com/intake/intake-parquet/blob/f029a36ba5c7a644b4faebd2a40d6ec21dbc5681/intake_parquet/source.py#L134 is actually a list of paths where the first one is the path to the parquet dir which is all that you actually need. This crops up when you run `df.persist()`

**Comment by jsignell** [comment]
Closing this for now, although I won't delete the branch yet.

**Pull request closed without merging**